src/graphql/admin/mutations.ts

```diff
@@ -1,34 +1,70 @@
-
 import AccountUpdateLevelMutation from "@graphql/admin/root/mutation/account-update-level"
 import AccountUpdateStatusMutation from "@graphql/admin/root/mutation/account-update-status"
+import AccountsAddUsdWalletMutation from "@graphql/admin/root/mutation/account-add-usd-wallet"
+import AccountCustomFieldsUpdateMutation from "@graphql/admin/root/mutation/account-custom-fields-update"
 import BusinessUpdateMapInfoMutation from "@graphql/admin/root/mutation/business-update-map-info"
 import ColdStorageRebalanceToHotWalletMutation from "@graphql/admin/root/mutation/cold-storage-rebalance-to-hot-wallet"
 
-import AccountsAddUsdWalletMutation from "./root/mutation/account-add-usd-wallet"
+import {
+  addAttributesToCurrentSpanAndPropagate,
+  SemanticAttributes,
+  ACCOUNT_USERNAME,
+} from "@services/tracing"
 
-const MutationType = GT.Object({
-  name: "Mutation",
-  fields: () => ({
-    userRequestAuthCode: UserRequestAuthCodeMutation,
-    userLogin: UserLoginMutation,
+const fields = {
+  userRequestAuthCode: UserRequestAuthCodeMutation,
+  userLogin: UserLoginMutation,
+
+  captchaCreateChallenge: CaptchaCreateChallengeMutation,
+  captchaRequestAuthCode: CaptchaRequestAuthCodeMutation,
 
-    captchaCreateChallenge: CaptchaCreateChallengeMutation,
-    captchaRequestAuthCode: CaptchaRequestAuthCodeMutation,
+  accountUpdateLevel: AccountUpdateLevelMutation,
+  accountUpdateStatus: AccountUpdateStatusMutation,
+  accountsAddUsdWallet: AccountsAddUsdWalletMutation,
 
-    accountUpdateLevel: AccountUpdateLevelMutation,
-    accountUpdateStatus: AccountUpdateStatusMutation,
-    accountsAddUsdWallet: AccountsAddUsdWalletMutation,
+  businessUpdateMapInfo: BusinessUpdateMapInfoMutation,
 
-    businessUpdateMapInfo: BusinessUpdateMapInfoMutation,
+  coldStorageRebalanceToHotWallet: ColdStorageRebalanceToHotWalletMutation,
+}
 
-    coldStorageRebalanceToHotWallet: ColdStorageRebalanceToHotWalletMutation,
-  }),
+const { customFields } = getAccountsConfig()
+if (customFields && customFields.length > 0) {
+  Object.assign(fields, { accountCustomFieldsUpdate: AccountCustomFieldsUpdateMutation })
+}
+
+const addTracing = () => {
+  for (const key in fields) {
+    // @ts-ignore-next-line no-implicit-any error
+    const original = fields[key].resolve
+    /* eslint @typescript-eslint/ban-ts-comment: "off" */
+    // @ts-ignore-next-line no-implicit-any error
+    fields[key].resolve = (source, args, context, info) => {
+      const { ip, domainAccount, domainUser } = context || {}
+      return addAttributesToCurrentSpanAndPropagate(
+        {
+          [SemanticAttributes.ENDUSER_ID]: domainUser?.id,
+          [ACCOUNT_USERNAME]: domainAccount?.username,
+          [SemanticAttributes.HTTP_CLIENT_IP]: ip,
+        },
+        // @ts-ignore-next-line no-implicit-any error
+        () => original(source, args, context, info),
+      )
+    }
+  }
+  return fields
+}
+
+const MutationType = GT.Object({
```

Comment:
there was no such tracing in mutation before?

Reply:
No but missing it in admin queries https://github.com/GaloyMoney/galoy/pull/1465/commits/969d2247d4e9da3054663dd7a280800526daa6cd